docs/src/scripts/extract_utils.py

```diff
@@ -0,0 +1,144 @@
+import re
```

Comment:
Missing copyright header.  Same for many other new files.
A file scope commenting describing the purpose of the script would be good.